### PR TITLE
[SPLIT] OrangeFox support

### PR DESCRIPTION
**Pull request by anon1892**
Some modifications to the code, so OrangeFox recovery can be supported (especially useful for Xiaomi devices)

Work started in #188 

It works, but I don't like the fix line 267 of `tooling.py`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,8 +48,9 @@ Linux is currently the best supported platform (tested with Ubuntu 20.04/22.04 L
       - [LineageOS for microg](https://download.lineage.microg.org)
       - [BlissRoms](https://blissroms.org)
       - [PixelExperience](https://download.pixelexperience.org)
-    - TWRP Recovery:
+    - some recoveries:
```
Comment (MagicLike) [inline README.md:51]: maybe change that to `Recoveries` :thinking:

Comment (anon1892) [reply README.md]: Yes, done. (was following same pattern as ROMs list)

**Comment by tsterbak** [comment]
I'm thinking about how to best proceed with these PRs.
I guess, I prepare a branch for release 0.5.0 and merge them one by one. This way I can make modifications to the code and you could test it with your devices. The only downside would be, that you would need to send a new PR to make changes. How does that sound for you?

**Comment by anon1892** [comment]
Yes, as you wish! Now I'm a pro at making PR, lol

**Comment by tsterbak** [comment]
I have a few questions going forward:
- Ofox recommends booting the recovery (at least for A/B-partitioned devices). I would also prefer that. What do you think? (source:  [can-i-run-fastboot-boot-to-boot-an-orangefox-img-file](https://wiki.orangefox.tech/en/guides/installing_orangefox#can-i-run-fastboot-boot-to-boot-an-orangefox-img-file)) So the process seems to be dangerous here...
-

**Comment by tsterbak** [comment]
> For non a/b devices, like lavender https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/222, I never do that (and not asked in LineageOS installation process)
I also created a function to boot the recovery, see rosemary https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/225 (I follow LineageOS instructions)

I see :) 

> Yes, we can also use the zip file, to check if it's the right OFox version. But after that, we need unzip it and only flash recovery.img file on it. It require more code. But can be a good option, because we will need to distinguish OFox with LOS recovery, which both are named recovery.img

Ah sorry, I missed, that it is a zip file...

**Pull request merged**